src/java/za/co/mip/ablduck/ABLDuck.java

```diff
@@ -0,0 +1,417 @@
+    
+            File outputFile = new File(this.destDirOutput, js.name + ".js");
+            try (FileWriter file = new FileWriter(outputFile.toString())) {
+                file.write("Ext.data.JsonP." + js.name.replace(".", "_") + "(" + gson.toJson(js) + ");");
+            } catch (IOException ex) {
+                throw new BuildException(ex);
+            }
+        }
+
+        File dataFile = new File(this.destDir, "data.js");
+        try (FileWriter file = new FileWriter(dataFile.toString())) {
+            file.write("Docs = {\"data\":" + gson.toJson(dataJSObject) + "}");
+        } catch (IOException ex) {
+            throw new BuildException(ex);
+        }
+        
+    }
+
+    private HierarchyResult determineClassHierarchy(SourceJSObject curClass, HierarchyResult result) {
+        
+        String inherits = curClass.ext;
+
+        if (!"".equals(inherits)) {
+            result.addHierarchy(inherits);
+
+            SourceJSObject nextClass = jsObjects.get(inherits);
+                        
+            if (nextClass != null) {
+                for (MemberObject member : nextClass.members) {
+                    if (member.owner.equals(inherits) && (member.meta.isPrivate == null || !member.meta.isPrivate)) 
+                        result.addInheritedmember(member);
+                }
+            
+                determineClassHierarchy(nextClass, result);
+            }
+        }
+        return result;
+    }
+
+    private void extractTemplateDirectory(File outputDir) throws IOException {
+ 
+        File template = exportResource("ablduck.zip", outputDir);
+
+        unzip(template.getAbsolutePath(), outputDir.getAbsolutePath());
+
+        template.delete();
+ 
+    }
+
+    public File exportResource(String resourceName, File outputDir) throws IOException {
+        InputStream stream = null;
+        OutputStream resStreamOut = null;
+        File resource;
+
+        try {
+            stream = ABLDuck.class.getResourceAsStream("resources/" + resourceName); 
+            if(stream == null) {
+                throw new IOException("Cannot get resource \"" + resourceName + "\" from Jar file.");
+            }
+    
+            int readBytes;
+            byte[] buffer = new byte[BUFFER_SIZE];
+            
+            resource = new File(outputDir, resourceName); 
+            resStreamOut = new FileOutputStream(resource.getAbsolutePath());
+            while ((readBytes = stream.read(buffer)) > 0) {
+                resStreamOut.write(buffer, 0, readBytes);
+            }
+        } catch (IOException ex) {
+            throw ex;
+        } finally {
+            if(stream != null)
+                stream.close();
+
+            if(resStreamOut != null)
+                resStreamOut.close();
+        }
+    
+        return resource;
+    }
+    
+    public void unzip(String zipFilePath, String destDirectory) throws IOException {
+        File destDir = new File(destDirectory);

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "destDir" which hides the field declared at line 92. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AHiddenFieldCheck)